src/openpgp/processor.iced

```diff
@@ -127,9 +119,15 @@ class KeyBlock
         continue
 
       n_sigs++
-      if (issuer_id = @_check_issuer(i, p, @primary))?
+      [issuer_id, matches_ours, expected_ours] = @_get_issuer i, p, @primary
+      if issuer_id?
+        unless matches_ours or p.type is C.sig_types.key_revocation
+          @warnings.push "Skipping signature by another issuer: #{issuer_id?.toString('hex')} != #{expected_ours?.toString('hex')}"
+          continue
+
         p.key = @primary.key
         p.primary = @primary
+        @opts.skip_verify = not matches_ours
```

Comment:
Can you make a copy of `@opts` and modify it and pass that in? This feels really risky. I don't want to turn verification off by accident.